examples/pruning/mxnet_integration.py

```diff
@@ -0,0 +1,124 @@
+    n_layers = trial.suggest_int('n_layers', 1, 3)
+
+    data = mx.symbol.Variable('data')
+    data = mx.sym.flatten(data=data)
+    for i in range(n_layers):
+        num_hidden = int(trial.suggest_loguniform('n_units_1{}'.format(i), 4, 128))
+        data = mx.symbol.FullyConnected(data=data, num_hidden=num_hidden)
+        data = mx.symbol.Activation(data=data, act_type="relu")
+
+    data = mx.symbol.FullyConnected(data=data, num_hidden=10)
+    mlp = mx.symbol.SoftmaxOutput(data=data, name="softmax")
+
+    return mlp
+
+
+def create_optimizer(trial):
+    # We optimize over the type of optimizer to use (Adam or SGD with momentum).
+    # We also optimize over the learning rate and weight decay of the selected optimizer.
+    weight_decay = trial.suggest_loguniform('weight_decay', 1e-10, 1e-3)
+    optimizer_name = trial.suggest_categorical('optimizer', ['Adam', 'MomentumSGD'])
+
+    if optimizer_name == 'Adam':
+        adam_lr = trial.suggest_loguniform('adam_lr', 1e-5, 1e-1)
+        optimizer = mx.optimizer.Adam(learning_rate=adam_lr, wd=weight_decay)
+    else:
+        momentum_sgd_lr = trial.suggest_loguniform('momentum_sgd_lr', 1e-5, 1e-1)
+        optimizer = mx.optimizer.SGD(momentum=momentum_sgd_lr, wd=weight_decay)
+
+    return optimizer
+
+
+def objective(trial):
+    # Generate trial model and trial optimizer.
+    mlp = create_model(trial)
+    optimizer = create_optimizer(trial)
+
+    # Load the test and train MNIST dataset.
+    mnist = mx.test_utils.get_mnist()
+    rng = np.random.RandomState(0)
+    permute_train = rng.permutation(len(mnist['train_data']))
+    train = mx.io.NDArrayIter(
+        data=mnist['train_data'][permute_train][:N_TRAIN_EXAMPLES],
+        label=mnist['train_label'][permute_train][:N_TRAIN_EXAMPLES],
+        batch_size=BATCHSIZE,
+        shuffle=True)
+    permute_test = rng.permutation(len(mnist['test_data']))
+    val = mx.io.NDArrayIter(
+        data=mnist['test_data'][permute_test][:N_TEST_EXAMPLES],
+        label=mnist['test_label'][permute_test][:N_TEST_EXAMPLES],
+        batch_size=BATCHSIZE)
+
+    # Create our MXNet trainable model and fit it on MNIST data.
+    model = mx.mod.Module(symbol=mlp)
+    model.fit(
+        train_data=train,
+        eval_data=val,
+        eval_end_callback=MXNetPruningCallback(trial, eval_metric='accuracy'),
+        optimizer=optimizer,
+        optimizer_params={'rescale_grad': 1.0 / BATCHSIZE},
+        num_epoch=EPOCH)
+
+    # Compute the accuracy on the entire test set.
+    test = mx.io.NDArrayIter(
+        data=mnist['test_data'],
+        label=mnist['test_label'],
+        batch_size=BATCHSIZE)
+    accuracy = model.score(eval_data=test, eval_metric='acc')[0]
+
+    return accuracy[1]
+
+
+if __name__ == '__main__':
+    import optuna
+    study = optuna.create_study(direction='maximize', pruner=optuna.pruners.MedianPruner())
+    study.optimize(objective, n_trials=100)
```

Comment:
Recently, we recommend specifying `timeout` argument in examples as follows (in order to let users know the presence of `timeout`):
```suggestion
    study.optimize(objective, n_trials=100, timeout=600)
```

Reply:
Thank you.

The motivation behind the use of `timeout` totally makes sense 👍